internal/http/services/appprovider/appprovider.go

```diff
@@ -512,3 +522,14 @@ func resolveViewMode(res *provider.ResourceInfo, vm string) gateway.OpenInAppReq
 	}
 	return viewMode
 }
+
+func appTargetToString(t apppb.Target) string {
```

Comment:
In case the the target is `invalid` you'll be opening it with `iframe`?

Reply:
Yeah, I thought about that, after all any future app provider would have to specify the target so let's not assume a default: I return empty string now (i.e. likely failing on the frontend)

Reply:
Not sure it makes much sense given the context (I'd have preferred a default value in protobuf, but it's an enum so it must have a value). But I don't have a strong opinion, so ok I changed it to `invalid`.